### PR TITLE
fix: revise link and improve information about topics in the contributing docs

### LINKED ISSUE #34378: Fix link and improve information about topics in the contributing docs
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/.github/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

https://docs.github.com/en/contributing/style-guide-and-content-model/about-topics

### What part(s) of the article would you like to see updated?

The first link in https://docs.github.com/en/contributing/style-guide-and-content-model/about-topics points to https://github.com/github/docs/tree/main/content#topics. It should point to https://docs.github.com/en/contributing/writing-for-github-docs/using-yaml-frontmatter#topics.

However, there is also scope to clarify what topics are and how they are defined in Markdown. The first paragraph of the article could be rewritten to explain that an article can be defined as being relevant to one or more topics by having those topics listed in the article's frontmatter - i.e. the metadata at the top of an article's Markdown file.

It would be more helpful to provide information about where to define the topics for an article, directly within the "About topics" article (with an example of a topics entry in the frontmatter), rather than linking away to other resources in the opening paragraph.

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by CBID2**
### Why:
This PR changes the link in the first paragraph in The "About Topics" article as well as adding an example on how to arrange the titles in frontmatter. Doing this would help contributors gain a better idea of how to contribute to GitHub's documentation.
### Closes: 
Closes #34378 

### Check off the following:

- [x] I have reviewed my changes in staging, available via the **View deployment** link in this PR's timeline (this link will be available after opening the PR).

  - For content changes, you will also see an automatically generated comment with links directly to pages you've modified. The comment won't appear if your PR only edits files in the `data` directory.
- [x] For content changes, I have completed the [self-review checklist](https://docs.github.com/en/contributing/collaborating-on-github-docs/self-review-checklist).

### CODE REVIEW

File: content/contributing/style-guide-and-content-model/about-topics.md [hunk 1]
```diff
@@ -6,7 +6,19 @@ versions:
...
+---
+title: "Managing branches in your repository"
+topics:
+  - "GitHub"
+  - "Git"
+  - "Repositories"
+---
+```
+
+For more information on adding topics to an article see, "[Topics](https://docs.github.com/en/contributing/writing-for-github-docs/using-yaml-frontmatter#topics)" and for a list of all allowed topics, see [`allowed-topics`](https://github.com/github/docs/blob/main/data/allowed-topics.js).
```
Comment (CBID2) [inline content/contributing/style-guide-and-content-model/about-topics.md:21]: @nguyenalex836, I'm not sure where the Markdown error is. 😕  The links both are formatted correctly.

Comment (nguyenalex836) [reply content/contributing/style-guide-and-content-model/about-topics.md:21]: @CBID2 That is a vexing error! 🤔 I'm going to see if our engineering team can speak to what might be causing it - I'll return when I have more info!

**Comment by nguyenalex836** [comment]
@CBID2 Thank you for opening this PR! ✨ While we wait for guidance on the [Markdown error](https://github.com/github/docs/pull/34491#discussion_r1744184217), I'll triage this for a writer to review the content you've added 💛

File: content/contributing/style-guide-and-content-model/about-topics.md (hunk 1 above)
Comment (nguyenalex836) [reply content/contributing/style-guide-and-content-model/about-topics.md]: Alrighty - so I believe I've fixed the first error, but I've run into another that I'll be getting help from the team on as well 😅 will update once I have more info!

Comment (nguyenalex836) [reply content/contributing/style-guide-and-content-model/about-topics.md]: @CBID2 Took a few tries, but I think we're set now ✨ will just need to wait for a writer to review the content!

**Comment by nguyenalex836** [comment]
@CBID2 I made a few minor updates regarding a run-on sentence and changing the phrase "Here's an example" to "For example" per guidance from our writers 💛  I'll update the branch and get this merged once tests are passing 🍏 

Thank you again for opening this PR! ✨

**Review by nguyenalex836 — APPROVED** [review]

**Pull request merged**